cowprotocol/cowswap
# [ETH] Stepper pure component

# Summary

This PR implements the pure component for the stepper. 

<img width="999" alt="image" src="https://user-images.githubusercontent.com/2352112/201962906-0634203e-7965-4555-a6f4-a33edbc0c971.png">

- Simple model for the parameters. Its easier to check in cosmos the parameters (its shown as a JSON in the fixture)
- Creates a component with the 3 steps, and 2 progress bars
- Step 1: Models the Send ETH tx state
- Step 2: Models the Order Creation state
- Step 3: Models the Receival of either the tokens or the refund

# Wireframes
- https://www.figma.com/proto/sUwumNFlcslHzShSZJHVsS/2022_COW-Native-ETH-Flow?page-id=1201%3A382&node-id=1232%3A1702&viewport=-96%2C1616%2C0.19&scaling=min-zoom&starting-point-node-id=1204%3A674
- Designs https://www.figma.com/proto/sUwumNFlcslHzShSZJHVsS/2022_COW-Native-ETH-Flow?page-id=1201%3A382&node-id=1298%3A382&viewport=-96%2C1616%2C0.19&scaling=min-zoom&starting-point-node-id=1298%3A382&show-proto-sidebar=1

## Cosmos cases
Cosmos allows to quickly check the happy path, but also many other not so obvious cases that can happen (i.e. the order expires before its creation).

<img width="650" alt="image" src="https://user-images.githubusercontent.com/2352112/201963261-a5eee71b-dd64-43a9-b386-f61f29ae205b.png">

## Not included
- I realised we don't have some global styles applied in the cosmos fixtures. I think we should. For example, to include the right fonts. This PR adds a hack to add the fonts in the fixture, but it should be done globally (follow up PR)
- Fine details and possible additional refactors, or restyling will be done by @fairlighteth 

# To Test

1. Go to Cosmos, open `EthFlowStepper` (`command + P` works best to find it)
     * https://pr1402--cowswap.review.gnosisdev.com/cosmos/?fixtureId=%7B%22path%22%3A%22src%2Fcow-react%2Fmodules%2Fswap%2Fpure%2FEthFlow%2FEthFlowStepper%2Findex.cosmos.tsx%22%7D 
2. Use the steps in the control to check how all the cases are handled
